### PR TITLE
#4076 - Queue Monitoring - Schedulers Refactor (SIN validation and loan balance)

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
- Refactored SIN validation-related schedulers and student loan balance import.
- Adjusted E2E tests.

### CODE REVIEW

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/sin-validation-integration/sin-validation-process-response-integration.scheduler.ts [hunk 1]
```diff
@@ -15,57 +17,35 @@ export class SINValidationResponseIntegrationScheduler extends BaseScheduler<voi
...
-      return {
-        processSummary: result.processSummary,
-        errorsSummary: result.errorsSummary,
-      };
-    });
+    return "ESDC SIN validation response files processed.";
   }
+
+  /**
+   * Logger for SFAS integration scheduler.
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/sin-validation-integration/sin-validation-process-response-integration.scheduler.ts:43]: Please adjust the comment.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/student-loan-balances/student-loan-balances-part-time-integration.scheduler.ts [hunk 2]
```diff
@@ -30,55 +26,35 @@ export class StudentLoanBalancesPartTimeIntegrationScheduler extends BaseSchedul
...
+    await this.studentLoanBalancesProcessingService.processStudentLoanBalances(
+      serviceProcessSummary,
+      auditUser.id,
+    );
+    processSummary.info("Completed processing Student Loan Balances files.");
+    return "Process finalized with success.";
   }
 
+  /**
+   * Logger for SFAS integration scheduler.
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/student-loan-balances/student-loan-balances-part-time-integration.scheduler.ts:52]: https://github.com/bcgov/SIMS/pull/4118/files#r1889415405

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/sin-validation-integration/sin-validation-process-response-integration.scheduler.ts [hunk 3]
```diff
@@ -15,57 +17,35 @@ export class SINValidationResponseIntegrationScheduler extends BaseScheduler<voi
...
-    });
-    await summary.info(
-      `Processing SIN validation integration job ${job.id} of type ${job.name}.`,
-    );
-    await summary.info("Processing ESDC SIN validation response files.");
-    const auditUser = this.systemUsersService.systemUser;
-    const results = await this.sinValidationProcessingService.processResponses(
-      auditUser.id,
-    );
-    await summary.info("ESDC SIN validation response files processed.");
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/sin-validation-integration/sin-validation-process-response-integration.scheduler.ts:60]: Can we have a `processSummary.info` with similar context ?

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/student-loan-balances/student-loan-balances-part-time-integration.scheduler.ts [hunk 4]
```diff
@@ -30,55 +26,35 @@ export class StudentLoanBalancesPartTimeIntegrationScheduler extends BaseSchedul
...
-      const errorMessage = "Unexpected error while executing the job.";
-      processSummary.error(errorMessage, error);
-      return [errorMessage];
-    } finally {
-      this.logger.logProcessSummary(processSummary);
-      await logProcessSummaryToJobLogger(processSummary, job);
-    }
+  protected async process(
+    _job: Job<void>,
+    processSummary: ProcessSummary,
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/student-loan-balances/student-loan-balances-part-time-integration.scheduler.ts:37]: Please add process summary to the comment.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/student-loan-balances/student-loan-balances-part-time-integration.scheduler.ts [hunk 5]
```diff
@@ -30,55 +26,35 @@ export class StudentLoanBalancesPartTimeIntegrationScheduler extends BaseSchedul
...
+  ): Promise<string> {
+    processSummary.info("Processing Student Loan Balances files.");
+    const auditUser = this.systemUsersService.systemUser;
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/student-loan-balances/student-loan-balances-part-time-integration.scheduler.ts:40]: May not be part of the PR. But the audit user seems to be not effective here and can be used directly in processing service.

**Comment by dheepak-aot** [comment]
Good job @andrewsignori-aot. Added a few minor comments.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/student-loan-balances/student-loan-balances-part-time-integration.scheduler.ts (hunk 5 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/student-loan-balances/student-loan-balances-part-time-integration.scheduler.ts:40]: Yeah, not really part of the PR but I am trying to change as possible. I will be making the change.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/sin-validation-integration/sin-validation-process-response-integration.scheduler.ts (hunk 3 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/sin-validation-integration/sin-validation-process-response-integration.scheduler.ts:60]: I am not sure about adding it. I see the point but the `BaseQueue` is already adding the below pair of logs to the `ProcessSummary` when the job starts and when it finishes
```
Processing queue ${job.queue.name}, job ID ${job.id}.
```
```
${job.queue.name}, job ID ${job.id}, executed.
```
_Note_: The base queue was wrongly adding the first log using the `logger` and the second one using the `ProcessSummary` and I just changed it in the last commit. Please take a look.

We need to decide if we need this "friendly" message at the processor level or the messages from the `BaseQueue` would be enough. Please lets have a quick chat and decide one way or another. I can go back and review the refactored schedulers.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/sin-validation-integration/sin-validation-process-response-integration.scheduler.ts (hunk 1 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/sin-validation-integration/sin-validation-process-response-integration.scheduler.ts]: Updated.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/student-loan-balances/student-loan-balances-part-time-integration.scheduler.ts (hunk 2 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/student-loan-balances/student-loan-balances-part-time-integration.scheduler.ts]: Updated.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/student-loan-balances/student-loan-balances-part-time-integration.scheduler.ts (hunk 4 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/student-loan-balances/student-loan-balances-part-time-integration.scheduler.ts:36]: Added.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/student-loan-balances/student-loan-balances-part-time-integration.scheduler.ts (hunk 5 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/student-loan-balances/student-loan-balances-part-time-integration.scheduler.ts]: Refactored.

Comment (dheepak-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/student-loan-balances/student-loan-balances-part-time-integration.scheduler.ts]: Thanks for the refactor.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/sin-validation-integration/sin-validation-process-response-integration.scheduler.ts (hunk 3 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/sin-validation-integration/sin-validation-process-response-integration.scheduler.ts:60]: As we talked, removing the friendly messages from the processor since the `BaseQueue` has some similar to indicate the start and end of the processing.

**Review by dheepak-aot — APPROVED** [review]
Thanks for making the changes. Looks good 👍
